annif/project.py

```diff
@@ -27,6 +28,7 @@ class AnnifProject(DatadirMixin):
     """Class representing the configuration of a single Annif project."""
 
     # defaults for uninitialized instances
+    _transformer = None
```

Comment:
I suggest renaming this field simply `_transform`, and all other entities (packages, classes, variables, functions...) with names like "transformer" similarly to just "transform", because Transformers are an established ML method and it might be confusing to have similarly named classes and variables that are not related to Transformers. In the future, Annif may even get one or more backends based on transformers (for example via pecos)